### ISSUE TITLE
Docs & UX: Better flakes recommendations?

### ISSUE THREAD

**Issue by doronbehar** [nix-community/home-manager#3967]
After years of Nix and NixOS experience, I tried `home-manager` for the first time as a declarative nix profile manager for a Debian machine with Nix (with daemon) installed on it.

I think that whenever one needs to setup such a machine, they should ideally need only these commands:

```
sh <(curl -L https://nixos.org/nix/install) --daemon # Or with `--no-daemon`
mkdir -p ~/.config/nix && echo 'extra-experimental-features = nix-command flakes' > ~/.config/nix/nix.conf
nix run home-manager/master -- switch --flake <flake-url>#<home-manager-profile>
```

However [home-manager's docs](https://nix-community.github.io/home-manager/index.html#ch-nix-flakes) suggest to use first the `init` sub command and then the `switch` subcommand and I'm not sure I understand why. If I have a repository ready to be used, why can't I run `switch` right away? The docs also don't mention that one can add a URL as an argument to `--flake` - it seems reasonable to me now, but it's not trivial IMO.

Additionally, the `init` subcommand always creates the `home.nix` file in the directory given to `--flake`, where [in my case](https://gitlab.com/doronbehar/nixos-configs) I specifically didn't create this file. This also convinced me that the `init` and `switch` subcommands won't accept `--flake <URL>` since otherwise the annoying creation of `home.nix` doesn't make sense if the flake is remote.

I also think it should be recommended to simply use `switch` because usually when one reaches such a machine `git` and `ssh` are not available - hence it's a bit harder to clone the "`nixos-configs`" repository, without the packages home-manager is supposed to install. This comes out as an issue if you `nix profile install nixpkgs#{git,openssh}` and then home-manager complains about a path collision with the packages installed imperatively via `nix profile install`.

One last thing: The docs also don't mention that a `#<home-manager-configuration-attrname>` argument is also supported - again I guessed it but it's worth mentioning. In my case, this works:

```
nix run home-manager/master -- switch --flake gitlab:doronbehar/nixos-configs#office
```

Here are some ideas:

- Maybe first explain how to use `switch --flake` thoroughly, and what the flake given needs to satisfy, including:
- - The `outputs` format, with examples that include multiple attributes in `homeConfigurations` and with more comments.
- - The supported `#...` flake URL argument, and the default `#...` derivation searched by `home-manager` (`#$USER` to my understanding).
- - Mention that flake URLs can be used as an argument to `switch --flake`.
- - Only after the above, mention the `init` subcommand which can help create such a directory.
- When `init` is used, don't create `home.nix` if a `flake.nix` exists - create both of them if `init --flake ...` is used and none of them exists, else create none.

Perhaps `home-manager init` should be totally removed, and be replaced with `nix flake init -t home-manager/master`, with explanations for the different variants.

**Comment by ghpzin** [comment]
It may warrant a separate issue, but `init` command appears to not respect home-manager branch when running in regards to inputs generated in flake.nix:
https://github.com/nix-community/home-manager/blob/a8d549351d4b87ab80665f35e57bee2a04201245/home-manager/home-manager#L395-L402

Meaning this part of the current docs may be misleading as both commands generate same inputs in flake.nix:
https://github.com/nix-community/home-manager/blob/33ccc5c28572d200f1e8d93fccba17e084e5f587/docs/nix-flakes.adoc?plain=1#L73-L88

**Comment by doronbehar** [comment]
I don't use home-manager anymore, so I will unsubscribe from this issue. It is still an issue however, so it would be nice to get this fixed.
